### PR TITLE
ROC-6845: Remove async feature flag

### PR DESCRIPTION
**Pull request by dharmendrak**
### JIRA link  ###

https://tools.hmcts.net/jira/browse/ROC-6845

### Change description ###

ROC-6845: Remove async feature flag

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/GetClaimOperation.java  3
         

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/documents/BulkPrintService.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/HMCTS/cmc-claim-store/pullRequest?prid=4479893&bid=15118134)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/GetClaimOperation.java  3
         

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/documents/BulkPrintService.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/HMCTS/cmc-claim-store/pullRequest?prid=4479893&bid=15118134)

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/GetClaimOperation.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    private ClaimRepository claimRepository;
+
+    @LogExecutionTime
+    @Retryable(value = RuntimeException.class, maxAttempts = 25, backoff = @Backoff(delay = 500))
+    public Claim getClaim(String externalId, String userAuthentication) throws Exception {
+
+        Optional<CaseMetadata> caseMetadata = Optional.of(retrieveCaseMetaData(externalId, userAuthentication));
+        if (!caseMetadata.filter(c -> c.getState() == ClaimState.OPEN).isPresent()) {
+            throw new RuntimeException("pin process not complete");
+        }
```
Comment (andrewwa-kainos) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/GetClaimOperation.java]: Can be simplified:
```
Optional.of(retrieveCaseMetaData(externalId, userAuthentication))
        .map(CaseMetadata::getState)
        .filter(ClaimState.OPEN::equals)
        .orElseThrow(() -> new RuntimeException("pin process not complete"));
```

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/GetClaimOperation.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    public Claim retrieveClaim(String externalId, String userAuthentication) throws Exception {
+        return claimRepository.getClaimByExternalId(externalId).get();
+    }
+
+    public CaseMetadata retrieveCaseMetaData(String externalId, String userAuthentication) throws Exception {
+        return claimRepository.getClaimByExternalId(externalId)
+            .map(CaseMetadata::fromClaim)
+            .get();
```
Comment (andrewwa-kainos) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/GetClaimOperation.java]: Not guaranteed to be present

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/GetClaimOperation.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import uk.gov.hmcts.cmc.claimstore.repositories.ClaimRepository;
+import uk.gov.hmcts.cmc.claimstore.stereotypes.LogExecutionTime;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.ClaimState;
+import uk.gov.hmcts.cmc.domain.models.metadata.CaseMetadata;
+
+import java.util.Optional;
+
+@Service
+public class GetClaimOperation {
```
Comment (sabahirfan) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/GetClaimOperation.java]: can we please rename this class?

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/ClaimOperation.java  3
         

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/documents/BulkPrintService.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/HMCTS/cmc-claim-store/pullRequest?prid=4479893&bid=15118134)

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/GetClaimOperation.java (hunk 2 above)
Comment (dharmendrak) [reply src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/GetClaimOperation.java]: actioned to return optional

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/ClaimOperation.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/documents/BulkPrintService.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/HMCTS/cmc-claim-store/pullRequest?prid=4479893&bid=15118134)

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/GetClaimOperation.java (hunk 1 above)
Comment (dharmendrak) [reply src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/GetClaimOperation.java]: simplified

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/PostClaimOperation.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/documents/BulkPrintService.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/HMCTS/cmc-claim-store/pullRequest?prid=4479893&bid=15118134)

**Pull request merged**